### DISCUSSION TITLE
Will CRIU's checkpoint/restore be introduce?

### DISCUSSION THREAD

**Discussion by AbdulR3hman** [docker-java/docker-java#2021] [Q&A]
Good Day everyone,

Just wondering if there is any appetite to introduce [docker checkpoint](https://docs.docker.com/engine/reference/commandline/checkpoint/) functionality at all? 

If not, any reasons to why? 

Kind regards,
AbdulRehman

**Comment by kiview** [comment]
I don't think there are strong plans to support this, as long as it is a highly experimental feature within Docker. Does it work on Docker Desktop for Mac and Windows nowadays?

**Reply by AbdulR3hman** [reply]
I believe so, it works well with Docker on Mac. It has been in experimental status for a while. I'm not sure why Docker still have it as experimental. Their first introduction of the feature -- [based on this blog](https://kubernetes.io/blog/2015/07/how-did-quake-demo-from-dockercon-work/) -- was around 2015. Is it the fact being experimental is the only reason for holding back? or generally speaking it is not a sought after? 

Thank you.

**Reply by AbdulR3hman** [reply]
Maybe it hasn't been in experimental status as long as I thought; the GitHub Docker ReadMe [1](https://github.com/docker/cli/commits/master/docs/reference/commandline/checkpoint.md)/[2](https://github.com/docker/cli/commits/aa89e6847a114e58fff3a0e758cc1abb1922203b/experimental/checkpoint-restore.md?browsing_rename_history=true&new_path=docs/reference/commandline/checkpoint.md&original_branch=master) suggest the feature was introduced in 2017... So not as long as 2015.

**Reply by kiview** [reply]
Thanks for the update regarding the state on Mac. I'll check out whether it is already supported on Windows as well. I assume you need to opt-in through experimental flag?

Can you elaborate on your use case for checkpoint support in `docker-java`? I think that would allow us to better prioritize this feature.

**Reply by AbdulR3hman** [reply]
Our need revolves around a java container that runs important executions, that needs to be warmed up, prepped, etc before it is able to accept requests. We'd like to be able to checkpoint the container after its all done with configurations and warming up and use it to replicate the container instead of creating a new one. 

The container "manager" so to speak is a micro-service that manages docker through `docker-java`.

**Reply by kiview** [reply]
Thanks for sharing, indeed this sounds like a good use case. Would you consider contributing this change to `docker-java`?
